corehq/apps/reports/management/commands/project_stats_report.py

```diff
@@ -339,8 +400,200 @@ def _get_table_size(config):
             for datasource in static_datasources + dynamic_datasources
         ], key=lambda r: r[-1] if r[-1] != 'Table not found' else 0)
 
+        ResourceModel.set_stat(ResourceModel.DATASOURCES_INFO, rows)
+
+    def _case_transactions(self):
+        db_name = get_db_aliases_for_partitioned_query()[0]
+        db_cursor = connections[db_name].cursor()
+
+        with db_cursor as cursor:
+            cursor.execute("""
+                SELECT COUNT(*) as num_forms, d.count as num_updates
+                FROM (
+                    SELECT COUNT(*) as count
+                    FROM form_processor_casetransaction
+                    WHERE case_id IN (
+                        SELECT case_id
+                        FROM form_processor_commcarecasesql
+                        WHERE domain = %s
+                    )
+                    GROUP BY form_id
+                ) AS d
+                GROUP BY d.count;
+            """, [self.domain])
+            result = cursor.fetchall()
+
+            running_form_case_updates = 0
+            total_forms = 0
+            for num_forms, num_updates in result:
+                total_forms += num_forms
+                running_form_case_updates += num_forms * num_updates
+
+            ResourceModel.set_stat(ResourceModel.CASE_TRANSACTIONS_FACTOR, running_form_case_updates / total_forms)
+
+            cursor.execute("""
+                SELECT COUNT(*)
+                FROM form_processor_casetransaction
+                WHERE case_id IN (
+                    SELECT case_id
+                    FROM form_processor_commcarecasesql
+                    WHERE domain = %s
+                );
+            """, [self.domain])
+            (total_transactions,) = cursor.fetchone()
+            ResourceModel.set_stat(ResourceModel.CASE_TRANSACTIONS_TOTAL, total_transactions)
+
+    def _case_indices(self):
+        db_name = get_db_aliases_for_partitioned_query()[0]
+        db_cursor = connections[db_name].cursor()
+
+        with db_cursor as cursor:
+            cursor.execute("""
+                SELECT COUNT(*)
+                FROM form_processor_commcarecaseindexsql
+                WHERE domain = %s;
+            """, [self.domain])
+            (total_case_indices,) = cursor.fetchone()
+
+            total_cases = ResourceModel.get_stat(ResourceModel.CASES_TOTAL)
+            if total_cases > 0:
+                ResourceModel.set_stat(ResourceModel.CASE_INDICES, total_case_indices/total_cases)
+
+    def _synclogs(self):
+        db_name = get_db_aliases_for_partitioned_query()[0]
+        db_cursor = connections[db_name].cursor()
+
+        with db_cursor as cursor:
+            cursor.execute("""
+                SELECT COUNT(*), d.count
+                FROM (
+                    SELECT COUNT(*) AS count
+                    FROM phone_synclogsql
```

Comment:
Mmm...no...I was not aware of the roll over, but I see the problem. But I'm actually thinking now...it seems there's a [configuration](https://github.com/dimagi/commcare-cloud/blob/a50ada52c88d22937577f088551bc1aac751a848/docs/commcare-cloud/env/postgresql_yml.md#synclogs) that allows a user to specify a different table to store synclogs. If I look at [this](https://github.com/dimagi/commcare-hq/blob/80fb62fe7bf9731f7cb6a614e8753a0b18e8983b/corehq/ex-submodules/casexml/apps/phone/tasks.py#L77) it seems there's maybe a different way I should approach the synclogs...maybe use something like `db = router.db_for_read(SyncLogSQL)`...

Reply:
That might have been a misleading code snippet. 
I mean that synclogs are stored in separate partitions based on the week 
https://github.com/dimagi/commcare-hq/blob/cd89af8efb71b8a178e999297bd65c8dca2ddd9b/corehq/ex-submodules/casexml/apps/phone/models.py#L425-L427
I would have thought to get the total number of synclogs, you would need to query each one of these tables. I think the alternative, as you say is to use the ORM to query it.

Reply:
Since we use use table inheritance it is possible to query just the parent table which will retrieve data from the child tables in the background.

Having said that it is likely that this data will be unreliable since we delete synclog data after a certain amount of time and also after successful syncs so I don't think it's possible to get the data you want from this table.

Reply:
@snopoke I presume you're referring to [this](https://github.com/dimagi/commcare-hq/blob/80fb62fe7bf9731f7cb6a614e8753a0b18e8983b/corehq/ex-submodules/casexml/apps/phone/tasks.py#L73)?

> Having said that it is likely that this data will be unreliable since we delete synclogs ... after successful syncs so I don't think it's possible to get the data you want from this table.

I guess the fact that synclogs is deleted after each successfully sync would result in the unreliability of the true number of synclogs produced by each user, but that's not the true aim here; the aim is rather the number of synclogs produced on average by the users that remain on the server to take up resources within the approx. 2 months before they are cleared.

I see we do consider the synclogs quite often as part of the resource model inputs, hence why I thought to include it as part of the output of the `project_stats_report` command...but I agree, the synclogs stat could vary vastly from month to month, but I'm not sure how else to report it?

Reply:
I was referring to this method that is called during form processing: https://github.com/dimagi/commcare-hq/blob/da986da1447bb23f42af0827b74c92dc1f98b844/corehq/ex-submodules/casexml/apps/case/util.py#L96

I think it's fine to include this data. It is useful in terms of DB disk but not as an indicator of sync frequency.